assets/src/domain/eventEditor/ui/tickets/ticketForm/useTicketFormConfig.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+import { format, parseISO } from 'date-fns';
+import { ProfileOutlined, CalendarOutlined, ControlOutlined } from '@ant-design/icons';
+
+import { defaultDateFormat, defaultTimeFormat } from '@appConstants/momentFormats';
+import { EspressoFormProps } from '@application/ui/forms/espressoForm';
+import useTicketItem from '../../../services/apollo/queries/tickets/useTicketItem';
+import { EntityId } from '@edtrServices/apollo/types';
+import { PLUS_ONE_MONTH, PLUS_TWO_MONTHS } from '../../../../shared/constants/defaultDates';
+import { processDateAndTime, DateAndTime } from '../../../../shared/services/utils/processDateAndTime';
+import { TicketBaseInput } from '@edtrServices/apollo/mutations';
+import TicketPriceCalculatorButton from '../ticketPriceCalculator/buttons/TicketPriceCalculatorButton';
+
+interface TicketFormShape extends TicketBaseInput, DateAndTime {
+	dateTime?: DateAndTime;
+}
+
+type TicketFormConfig = EspressoFormProps<TicketFormShape>;
+
+const useTicketFormConfig = (id: EntityId, config?: EspressoFormProps): TicketFormConfig => {
+	const { name, description, quantity, startDate: start, endDate: end } = useTicketItem({ id }) || {};
+
+	const startDate = start ? parseISO(start) : PLUS_ONE_MONTH;
+	const endDate = end ? parseISO(end) : PLUS_TWO_MONTHS;
+
+	const { onSubmit } = config;
+
+	const onSubmitFrom: TicketFormConfig['onSubmit'] = ({ dateTime, ...rest }, form, ...restParams) => {
+		// convert "dateTime" object to proper "startDate" and "endDate"
+		const { startDate, endDate } = processDateAndTime(dateTime);
+
+		const values = { ...rest, startDate, endDate };
+
+		return onSubmit(values, form, ...restParams);
+	};
+
+	const adjacentFormItemProps = {
+		labelCol: { span: 10 },
+		wrapperCol: { span: 14 },
+		labelAlign: 'left' as 'left',
+		style: {
+			width: '49%',
+			display: 'inline-flex',
```

Comment:
we'll need to improve this later or set these as default and allow to be overriden

Reply:
Agreed, that's why I added that `className`